client/finality-grandpa/src/lib.rs

```diff
@@ -552,6 +568,13 @@ where
 
 	let (voter_commands_tx, voter_commands_rx) = tracing_unbounded("mpsc_grandpa_voter_command");
 
+	// Q: Is there any way to enforce that they point to the same set of notifiers?
```

Comment:
Yes, instead of creating them like this, always create them as a pair and disallow constructing individual halfs.
```
let (sender, receiver) = GrandpaJustifications::channel();
```